libkbfs/md_ops.go

```diff
@@ -55,11 +55,25 @@ func init() {
 type MDOpsStandard struct {
 	config Config
 	log    logger.Logger
+
+	lock sync.Mutex
+	// For each TLF, maps an MD revision representing the next MD
+	// after a device revoke, with the minimum revision number that's
+	// been validated in chain up to the given MD revision.  That is,
+	// for TLF 1, if we have a next revision of 1000, and we've
+	// validated that MDs 100-1000 form a valid chain, then the map
+	// would contain: {1: {1000: 100}}
+	leafChainsValidated map[tlf.ID]map[kbfsmd.Revision]kbfsmd.Revision
```

Comment:
Is the idea that we are always checking on the revision right after a device revoke?

Reply:
> Is the idea that we are always checking on the revision right after a device revoke?

Yep.  When we see something signed by a revoked key, we ask the service what revision came back right after the revoke, and always check that one.